packages/plugin-mcp/src/endpoints/mcp.ts

```diff
@@ -13,43 +13,60 @@ export const initializeMCPHandler = (pluginOptions: PluginMCPServerConfig) => {
     const MCPHandlerOptions = MCPOptions.handlerOptions || {}
     const useVerboseLogs = MCPHandlerOptions.verboseLogs ?? false
 
-    const apiKey = req.headers.get('Authorization')?.startsWith('Bearer ')
-      ? req.headers.get('Authorization')?.replace('Bearer ', '').trim()
-      : null
+    let mcpAccessSettings: MCPAccessSettings
 
-    if (apiKey === null) {
-      throw new APIError('API Key is required', 401)
-    }
+    if (pluginOptions.overrideAuth) {
+      try {
+        mcpAccessSettings = await pluginOptions.overrideAuth(req)
+        if (useVerboseLogs) {
+          payload.logger.info('[payload-mcp] Custom authentication method successful')
+        }
+      } catch (error) {
+        payload.logger.error({
+          err: error,
+          msg: '[payload-mcp] Custom authentication method failed',
+        })
+        throw new APIError('Authentication failed', 401)
+      }
+    } else {
```

Comment:
FWIW, you could remove the if/else block and instead move the logic from this `else` to a function that gets assigned as a default to `overrideAuth`.

This way we can export it and users could import and call it in their own override as a way to extend existing functionality by adding code before/after calling the default.

Reply:
I like that! Will do!